nixos/modules/services/x11/display-managers/gdm.nix

```diff
@@ -229,40 +229,15 @@ in
 
     systemd.user.services.dbus.wantedBy = [ "default.target" ];
 
-    programs.dconf.profiles.gdm =
-    let
-      customDconf = pkgs.writeTextFile {
-        name = "gdm-dconf";
-        destination = "/dconf/gdm-custom";
-        text = ''
-          ${optionalString (!cfg.gdm.autoSuspend) ''
-            [org/gnome/settings-daemon/plugins/power]
-            sleep-inactive-ac-type='nothing'
-            sleep-inactive-battery-type='nothing'
-            sleep-inactive-ac-timeout=0
-            sleep-inactive-battery-timeout=0
-          ''}
-        '';
-      };
-
-      customDconfDb = pkgs.stdenv.mkDerivation {
-        name = "gdm-dconf-db";
-        buildCommand = ''
-          ${pkgs.dconf}/bin/dconf compile $out ${customDconf}/dconf
-        '';
+    programs.dconf.profiles.gdm.databases = [ "${gdm}/share/gdm/greeter-dconf-defaults" ]
+      ++ lib.optionals cfg.gdm.autoSuspend [{
+      settings.org.gnome.settings-daemon.plugins.power = {
+        sleep-inactive-ac-type = "nothing";
+        sleep-inactive-battery-type = "nothing";
+        sleep-inactive-ac-timeout = lib.gvariant.mkInt32 0;
+        sleep-inactive-battery-timeout = lib.gvariant.mkInt32 0;
       };
-    in pkgs.stdenv.mkDerivation {
-      name = "dconf-gdm-profile";
-      buildCommand = ''
-        # Check that the GDM profile starts with what we expect.
```

Comment:
Do you mean that I should add the profile content into passthru then check it in the module?

Reply:
I meant adding the profile content into `passthru`:

```nix
passthru = {
  dconfDb = "${finalAttrs.finalPackage}/share/gdm/greeter-dconf-defaults";
  dconfProfile = ''
    user-db:user
    file-db:${finalAttrs.passthru.dconfDb}
  '';
};
```

checking it in `passthru.tests`, and using `gdm.dconfDb`  in the NixOS module.

It will require switching the package to `finalAttrs` first (see e.g. https://github.com/NixOS/nixpkgs/pull/237140).